Source/Core/Core/Scripting/HelperClasses/GCButton_API_Implementations.cpp

```diff
@@ -0,0 +1,292 @@
+#include "Core/Scripting/HelperClasses/GCButton_API_Implementations.h"
+
+#include "Core/Scripting/CoreScriptInterface/Enums/GCButtonNameEnum.h"
+
+#include "string.h"
+
+namespace Scripting
+{
+
+static bool IsEqualIgnoreCase(const char* string_1, const char* string_2)
+{
+  int index = 0;
+  while (string_1[index] != '\0')
+  {
+    if (string_1[index] != string_2[index])
+    {
+      char first_char = string_1[index];
+      if (first_char >= 65 && first_char <= 90)
+        first_char += 32;
+      char second_char = string_2[index];
+      if (second_char >= 65 && second_char <= 90)
+        second_char += 32;
+      if (first_char != second_char)
+        return false;
+    }
+    ++index;
+  }
+
+  return string_2[index] == '\0';
+}
+
+int ParseGCButton_impl(const char* button_name)
+{
+  if (strlen(button_name) == 1)
+  {
+    switch (button_name[0])
+    {
+    case 'a':
+    case 'A':
+      return (int)GCButtonNameEnum::A;
+
+    case 'b':
+    case 'B':
+      return (int)GCButtonNameEnum::B;
+
+    case 'x':
+    case 'X':
+      return (int)GCButtonNameEnum::X;
+
+    case 'y':
+    case 'Y':
+      return (int)GCButtonNameEnum::Y;
+
+    case 'z':
+    case 'Z':
+      return (int)GCButtonNameEnum::Z;
+
+    case 'l':
+    case 'L':
+      return (int)GCButtonNameEnum::L;
+
+    case 'r':
+    case 'R':
+      return (int)GCButtonNameEnum::R;
+
+    default:
+      return (int)GCButtonNameEnum::UnknownButton;
+    }
+  }
+
+  switch (button_name[0])
```

Comment:
This doesn't verify that `button_name` is actually set. You check for length 1 earlier, but never for 0.

And since this is user-controlled, it should be a bit more sturdy than that (unless I missed an earlier place that doesn't let you call this with an empty string)

Reply:
Fixed.